bittide-instances/src/Bittide/Instances/Hitl/HwCcTopologies.hs

```diff
@@ -488,54 +632,141 @@ hwCcTopologyTest refClkDiff sysClkDiff syncIn rxns rxps miso =
 
 tests :: HitlTestsWithPostProcData TestConfig SimConf
 tests = Map.fromList
-  [ testTopology   diamond
-  , testTopology $ complete d3
-  , testTopology $ cyclic d5
-  , testTopology $ torus2d d2 d3
-  , testTopology $ star d7
-  , testTopology $ line d4
-  , testTopology $ hourglass d3
+  [ -- CALIBRATION --
+    -----------------
+
+    -- detect the natual clock offsets to be elided from the later tests
+    calibrateClockOffsets
+
+    -- TESTS --
+    -----------
+
+    -- initial clock shifts   startup delays            topology
+  , tt icsDiamond             ((m *) <$> sdDiamond)     diamond
+  , tt icsComplete            ((m *) <$> sdComplete)  $ complete d3
+  , tt icsCyclic              ((m *) <$> sdCyclic)    $ cyclic d5
+  , tt icsTorus               ((m *) <$> sdTorus)     $ torus2d d2 d3
+  , tt icsStar                ((m *) <$> sdStar)      $ star d7
+  , tt icsLine                ((m *) <$> sdLine)      $ line d4
+  , tt icsHourglass           ((m *) <$> sdHourglass) $ hourglass d3
   ]
  where
+  m = 1_000_000
+
+  icsDiamond = -1000 :> -500 :> 2000 :> 3000 :> Nil
+  sdDiamond  =     0 :>   10 :>  200 :>    3 :> Nil
```

Comment:
These are integration tests, which is why they should be deterministic and not random. We should not start mixing experiments with testing our infrastructure. Instead, we should set up dedicated experiments to run the parameter sweep (as this has nothing to do with CI). Doing so for the simulation runs was just a workaround to get some first insights without introducing to much infrastructure, but that is no longer a high-yielded path IMO.

Also note the that tests would fail in many cases if using random numbers at the moment. Our selected parameter set is still to fragile to be universal enough for the full parameter range.

Reply:
Gotcha makes sense!